SG14_test/ring_test.cpp

```diff
@@ -175,19 +175,19 @@ static void reverse_iterator_test()
     r.push_back(2);
     r.push_back(3);
     r.push_back(4);
-    std::vector<int> v(3);
+    std::vector<double> v(3);
```

Comment:
I think this change is incorrect, or at least, significant. The old code tested that `std::copy` from `ring<double>` to `vector<int>` worked; the new code tests only from `ring<double>` to `vector<double>` (i.e., between iterators sharing the same `value_type`).

I think the compiler is being overzealous here. However, would it be satisfactory (to both you and the compiler) to switch the types so that we're copying from a `ring<int>` into a `vector<double>`, or maybe from `ring<char>` into `vector<int>`?